### PR TITLE
use a binary ledger: newline-separated, newline-escaped entries instead of json

### PR DESCRIPTION
**Pull request by rob-solana**

### CODE REVIEW

File: src/entry_writer.rs [hunk 1]
```diff
@@ -19,8 +19,25 @@ impl<'a, W: Write> EntryWriter<'a, W> {
...
+            if c == b'\\' || c == b'\n' {
+                trace!("escaping '{}' to {}", c, c + b'\\');
+                escaped.push(b'\\');
+                c += b'\\';
+            }
+            escaped.push(c);
+        }
+        trace!("write_entry escaped = {:?}", escaped);
+
+        writer.write(&escaped[..])?;
```
Comment (mvines) [inline src/entry_writer.rs]: I still don't understand why the binary format couldn't be tuples of `(raw.len(),raw)` with no escaping.

Comment (rob-solana) [reply src/entry_writer.rs]: good idea, I'll try that approach

Comment (rob-solana) [reply src/entry_writer.rs]: most of the trouble with the len->bytes approach is my feebleness with rust...  constructing an iterator from something that's calling read() read()...

**Comment by garious** [comment]
Can you update the description with more context? Are you optimizing for space? For speed? Why now? What ticket will this close?

**Comment by garious** [comment]
@rob-solana, close this one in favor of your other PR?

**Comment by rob-solana** [comment]
subsumed by other binary PR

**Pull request closed without merging**